python/ray/rllib/optimizers/multi_gpu_optimizer.py

```diff
@@ -35,16 +35,16 @@ def _init(self,
               sgd_stepsize=5e-5,
               num_sgd_iter=10,
               timesteps_per_batch=1024,
+              num_gpus=0,
               standardize_fields=[]):
         self.batch_size = sgd_batch_size
         self.sgd_stepsize = sgd_stepsize
         self.num_sgd_iter = num_sgd_iter
         self.timesteps_per_batch = timesteps_per_batch
-        gpu_ids = ray.get_gpu_ids()
-        if not gpu_ids:
+        if not num_gpus:
             self.devices = ["/cpu:0"]
         else:
-            self.devices = ["/gpu:{}".format(i) for i in range(len(gpu_ids))]
+            self.devices = ["/gpu:{}".format(i) for i in range(num_gpus)]
```

Comment:
Did the prior approach work? Shouldn't `self.devices` always be indexed from `0` to `num_gpus-1`? `ray.get_gpu_ids` are used under the hood to set `CUDA_VISIBLE_DEVICES`, but I wouldn't expect those numbers to be meaningful in this context.

Reply:
Oh, `for i in range(len(gpu_ids))` doesn't actually use the IDs of the GPUs

Reply:
Oh I see.

Reply:
Yeah tensorflow always indexes those devices sequentially.